share/man/man9/cdefs.9

```diff
@@ -0,0 +1,297 @@
+.\"-
+.\" Copyright (c) 2024 M. Warner Losh <imp@FreeBSD.org>
+.\"
+.\" SPDX-License-Identifier: BSD-2-Clause
+.\"
+.Dd July 3, 2024
+.Dt CDEFS 9
+.Os
+.Sh NAME
+.Nm cdefs
+.Nd "Compiler portability definitions"
+.Sh DESCRIPTION
+.In sys/cdefs.h
+defines macros for compiler, C language standard portability, POSIX standards
+compliance and symbol visibility.
+It defines programming interfaces for the system header files to adopt to the
+many environments
+.Fx
+supports compilation for.
+It defines convenience macros for the
+.Fx
+sources, tailored to the base
+system's needs.
+.Pp
+Most of these macros are for use inside the
+.Fx
+sources only.
+They are not intended as a general portability layer.
+.Sh Supported Compilers
+There are two levels of compiler support for
+.Fx :
+Compilers supported for building programs on
+.Fx .
+These are compilers that you can use to build and run programs on
+Fx .
+.Bl -column -offset indent ".Sy Compiler" ".Sy Versions"
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 9, 10, 11, 12, 13, 14
+.It clang Ta 10, 11, 12, 13, 14, 15, 16, 17, 18
+.It TinyC (tcc) Ta 0.9
+.It pcc Ta 1.1
+.El
+Due to testing constraints, tcc and pcc may not always work.
+.PP
+Compilers supported for building
+.Fx
+itself:
+.Bl -column -offset indent ".Sy Compiler" ".Sy Versions"
+.It Sy Compiler Ta Sy Versions
+.It gcc Ta 12, 13
+.It clang Ta 16, 17, 19
+.El
+.Sh Macros and Magic for Programming Environment
+.Nm
+defines (or refrains from defining) a number of macros to increase portability
+of compiled programs.
+These are to allow more advanced language features to appear in header files.
+The header files assume a compiler that accepts C prototype function
+declarations.
+They also assume bare ansi keywords can be used.
+The
+.Dv __P
+macro is deprecated, will be removed in the future and not even documented.
+.Ss General Macros
+.Bl -column ".Sy Macro" ".Sy Description"
+.It Sy Macro Ta Sy Description
+.It Dv __const Ta expands to const in ansi and newer environments,
+__const in pre-ansi environments that support this extension or nothing
+otherwise.
+This my be removed soon, since it is unused in our tree.
+.It Dv __signed Ta expands to signed in ansi and newer environments,
+__signed in pre-ansi environments that support this extension or nothing
+otherwise.
```

Comment:
```suggestion
.It Dv __signed Ta expands to signed in ANSI and newer environments,
__signed in pre-ANSI environments that support this extension, otherwise
contracts to nothing.
```

Reply:
I've removed __signed from cdefs.h, so will remove it here instead.